### DISCUSSION TITLE
Running an application for beginnners?

### DISCUSSION THREAD

**Discussion by mmaywald05** [beehive-lab/TornadoVM#188] [Q&A] [answered]
Hello,

Im came across TornadoVM and want to try it out. I am very much new to the whole ordeal, so I thank you in advance for your patience with me.
I can't manage to start simple applcations, even after reading through the material and discussion #107.
I installed it for Windows with Graal following the [doc ](https://github.com/beehive-lab/TornadoVM/blob/master/assembly/src/docs/20_INSTALL_WINDOWS_WITH_GRAALVM.md)and can list  available devices and run the included examples perfectly fine.
What I can't manage to do is run an own application, seemingly no matter the content. I would be very thankful for some explicit directions. 

TornadoVM's files are located at C:/Projects/TornadoVM, my testing app at C:/Projects/TornadoTest. The code for now is just a copy of the MatrixMultiplication2D class, which again works perfectly fine when called from the tornado-examples package. 

How would I execute the app using the MSys2 shell? As in, what is the actual process/command to execute an App from a different (sister-)directory? Everything I tried myself lead to exceptions seemingly caused by wrong path calls. 

This is the directory structure:

C:/Projects
> TornadoVM
> 
>> assembly
>> benchmarks
>> tornado-examples
>>> src/main/java/uk/ac/manchester/tornado/examples/compute/MatrixMultiplication2D.java
>>
>>...
>
> TornadoTest
>>
>> src/main/java
>>>  MatrixMultiplication2D.java
>>
>> target
>> .idea
>> pom.xml

So, what's the command for executing this test? Do i use the TornadoVM install directory in MSys2 for this command? And is there anything else I need to do to the test (aside from including the maven dependencies)?

Thank you very much for your help, if there are any questions that will help my issue I will be happy to answer them.

**Comment by jjfumero** [comment, marked as answer]
Hi @mmaywald05 , 
   thank you for the report. To run standalone applications, we recommend to set up a maven project. You can compile with the `mvn` command as usual, and then run your application with the `tornado` command. Note that `tornado` is just an alias for your $JAVA_HOME + all flags to enable TornadoVM (Java modules, and paths to the native libraries for OpenCL, PTX and Level Zero).

We recommend starting with one of our templates, for example, this one:
https://github.com/jjfumero/tornadovm-examples 

Let's know if this helps.

**Reply by mmaywald05** [reply]
Thank you, that actually did it for me. The linked example helped a lot.
